### PR TITLE
fix : Wrong information in the docs about run-sh and other `io.fabric8` artifacts (#1006)

### LINKED ISSUE #1006: Wrong information in the docs about run-sh and other io.fabric8 artifacts
### Info

I was looking at:
* https://www.eclipse.org/jkube/docs/kubernetes-maven-plugin
* https://www.eclipse.org/jkube/docs/openshift-maven-plugin

and I believe these are the transitioned versions of https://dmp.fabric8.io/#maven-goals

For example in the fabric8's docker-maven-plugin there's:
```
Build Plugins

This plugin supports so call dmp-plugins which are used during the build phase. dmp-plugins are enabled by just declaring a dependency in the plugin declaration:

<plugin>
  <groupId>io.fabric8</groupId>
  <artifactId>docker-maven-plugin</artifactId>

  <dependencies>
    <dependency>
      <groupId>io.fabric8</groupId>
      <artifactId>run-java-sh</artifactId>
      <version>1.2.2</version>
    </dependency>
  </dependencies>
</plugin>

These plugins contain a descriptor META-INF/maven/io.fabric8/dmp-plugin with class names, line-by-line:

io.fabric8.runsh.RunShLoader
```

While in jkube docs there's simply:
```
Build Plugins

This plugin supports so call dmp-plugins which are used during the build phase. dmp-plugins are enabled by just declaring a dependency in the plugin declaration:

<plugin>
  <groupId>org.eclipse.jkube</groupId>
  <artifactId>kubernetes-maven-plugin</artifactId>

  <dependencies>
    <dependency>
      <groupId>org.eclipse.jkube</groupId>
      <artifactId>run-java-sh</artifactId>
      <version>1.2.2</version>
    </dependency>
  </dependencies>
</plugin>

org.eclipse.jkube.runsh.RunShLoader
```

while `kubernetes-maven-plugin` artifact _was_ transitioned from `io.fabric8` to `org.eclipse.jkube` groupId, it isn't the case with `io.fabric8/run-java-sh` GA and `io.fabric8.runsh.RunShLoader` class...

### PR DESCRIPTION
**Pull request by rohanKanojia**
## Description

Fix #1006

+ Fix outdated dmp plugin descriptor for build-plugins functionality. Introduce an interface `org.eclipse.jkube.kit.api.JKubeBuildPlugin` which would implemented by build plugins
+ rather than relying on on reflection, add an interface
    `org.eclipse.jkube.kit.common.JKubeBuildPlugin` which would be
    implemented by each build plugin. Instead of loading manually, use
    `java.util.ServiceLoader` to load the implementations automatically.
+ Quickstart added in https://github.com/eclipse/jkube/pull/1711
+ Docs added in https://github.com/eclipse/jkube/pull/1712

Signed-off-by: Rohan Kumar <rohaan@redhat.com>

## Type of change

 - [X] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [X] I have read the [contributing guidelines](https://www.eclipse.org/jkube/contributing)
 - [X] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [X] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [X] I have implemented unit tests to cover my changes
 - [X] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [X] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Comment by rohanKanojia** [comment]
For now it's just a simple implementation to give a clear idea about how this functionality works. We can iterate over it to make it as a more sensible use case.

File: jkube-kit/doc/src/main/asciidoc/inc/build/_buildargs.adoc [hunk 1]
```diff
@@ -2,16 +2,31 @@
...
+
+.Example
+[source,properties,indent=0,subs="verbatim,quotes,attributes"]
+----
+docker.buildArg.myBuildArg = myValue
+----
+endif::[]
+ifeval::["{plugin-type}" == "maven"]
+* Set a system property when running {plugin-type}, e.g: `-Ddocker.buildArg.http_proxy=http://proxy:8001`. This is especially
+useful when using predefined Docker arguments for setting proxies transparently.
```
Comment (sunix) [inline jkube-kit/doc/src/main/asciidoc/inc/build/_buildargs.adoc:18]: Minor: these 2 lines are common to maven and gradle, right? could these be out of the ifeval blocks?

Comment (rohanKanojia) [reply jkube-kit/doc/src/main/asciidoc/inc/build/_buildargs.adoc:18]: Thanks for noticing! Let me update it

**Comment by manusa** [comment]
Let's create an interface so we don't have to rely on duck types and reflection

**Comment by sunix** [comment]
I feel like this PR is too big. Could it be splitted?

**Comment by rohanKanojia** [comment]
@sunix : I've split this PR into 3 PRs:

Code For JKube Build Plugins : https://github.com/eclipse/jkube/pull/1525
JKube Build Plugin Quickstart: https://github.com/eclipse/jkube/pull/1711
Documentation : https://github.com/eclipse/jkube/pull/1712

**Review by manusa — APPROVED** [review]

**Pull request merged**